### PR TITLE
fix: use eslint linter instead of the cli engine

### PR DESCRIPTION
**Pull request by varl**
This has the primary benefit of letting this instance of ESLint run in isolation from the [ESLint configuration cascade and hierarchy](https://eslint.org/docs/user-guide/configuring#configuration-cascading-and-hierarchy) which can interfere with these standards.

Using only the linter and dealing with the output explicitly allows a project to have a local instance of ESLint for e.g. React 16.8 linting and cli-style for complying with DHIS2 code standards.

This has a secondary benefit of being faster. For maintenance app a full check went from **9.8 sec** to **7.2 sec**.

### CODE REVIEW

File: src/tools/js/eslint.js [hunk 1]
```diff
@@ -21,32 +23,27 @@ module.exports = (file, text, apply = false) => {
...
     try {
-        const cli = new eslint.CLIEngine({
-            configFile: eslintConfig,
-            useEslintrc: false,
-            fix: apply,
+        const { messages, fixed, output } = linter.verifyAndFix(text, eslintConfig, {
+            filename: path.basename(file),
             allowInlineConfig: true,
-            ignore: false,
+            reportUnusedDisableDirectives: false,
```
Comment (amcgee) [inline src/tools/js/eslint.js]: What's the reason for this change?

Comment (varl) [reply src/tools/js/eslint.js]: The `ignore` looks like it got removed because of the way Git shows the diff. The `ignore` prop is actually for the configuration object for `CLIEngine`. The configuration property `ignore` does not exist for `Linter.verifyAndFix`.

I initially set `reportUnusedDisableDirectives` to `true` as it is quite handy to know if the `/* eslint-disable */` pragma is necessary or not. But this would only check if the pragma is required according to the ESLint configuration in `cli-style` whereas it is required by the project-level ESLint configuration, so I explicitly  set it to `false` to signal that it is `false` for a reason.

**Comment by varl** [comment]
That self-test you put in the Travis file is a life-saver!
